src/formats/anvil2/anvil2_world.py

```diff
@@ -238,12 +238,8 @@ def get_blocks(self, cx: int, cz: int) -> Union[numpy.ndarray, NotImplementedErr
         # ]  # Get all unique blocks for all the sections and remove air
         for unique in uniques:
             internal = self._materials.get_block_from_definition(unique, default=unique)
-            internal_in_mapping = numpy.where(self.mapping_handler == internal)[0]
-            if len(internal_in_mapping) > 0:
-                internal_id = internal_in_mapping[0]
-            else:
-                internal_id = len(self.mapping_handler)
-                self.mapping_handler = numpy.append(self.mapping_handler, internal)
+            block: Block = self.get_blockstate(internal)
+            internal_id = self.mapping_handler[block]
```

Comment:
Same issue as above:
It's really implicit here that it first sets the block_object into mapping_handler and then finds its location. The part that adds block_object into mapping_handler should explicit and probably on a separate line. So it will first add it to mapping_handler and only then find the internal_id.